src/lib/support/ScopedBuffer.h

```diff
@@ -130,9 +132,15 @@ class ScopedMemoryBuffer : public Impl::ScopedMemoryBufferBase<MemoryManagement>
 public:
     using Base = Impl::ScopedMemoryBufferBase<MemoryManagement>;
 
+    static_assert(std::is_trivial<T>::value, "Constructors won't get run");
+
     inline T * Ptr() { return static_cast<T *>(Base::Ptr()); }
+    inline operator T *() { return Ptr(); }
```

Comment:
I think this implicit conversion hurts readability. It makes passing a pointer look the same as passing by value.

- `std::unique_ptr` doesn't have such a conversion.
- Having an explicit `get()` or `Ptr()` is in line with style principles used at Google to optimize for readers over writers. The function call is a hint that a conversion is happening (it's not a trivial one).

Reply:
Hmm.  That's a good point.

@mspang If you're ok with the `operator[]` bits, which should reduce the need for `Ptr()` usage when writing to or reading from the buffer locally, then it makes sense to me to require an explicit call to get a "borrowed" pointer.

We could keep calling it `Ptr`, or call it `Get` or `get` to align with `std::unique_ptr`, or perhaps call it `Borrow()` to make the ownership semantics more explicit.  Any preferences?

Reply:
I'd weakly prefer the same name as unique_ptr.

I don't mind the `operator[]` and that's already consistent with unique_ptr<T[]>.

Reply:
So weak preference for `get` or weak preference for `Get`, which is more CHIP-style?

Reply:
From my view, we should align with CHIP style and use `Get` since we are not generally embracing STL in CHIP.